pkg/operator/ceph/cluster/cephstatus.go

```diff
@@ -195,3 +197,43 @@ func (c *ClusterController) updateClusterCephVersion(image string, cephVersion c
 		return
 	}
 }
+
+// updateCephStorage queries the device class information and udates the CR
+func (c *ClusterController) updateCephStorage() {
+	var err error
+	var devices []string
+	retryCount := 5
+	cephCluster := &cephv1.CephCluster{}
+	cephClusterStorage := cephv1.CephStorage{}
+
+	//wait a bit longer to get osd details reflected in the crush map
+	for i := 0; i < retryCount; i++ {
+		devices, err = cephclient.GetDeviceClasses(c.context, c.namespacedName.Namespace)
+		if err != nil {
+			logger.Errorf("failed to get storage device classes. %v", err)
```

Comment:
This assumes the command will eventually succeed. I'm just not a big fan of waiting in general especially given that depending on the size of the cluster we might need more time. Perhaps this should be part of another go routine that runs at regular intervals and tries to populate the Status field.

Reply:
I had hoped that we would only need to check this once at the end of a reconcile since that is the only time we possibly expect a change in the CRUSH map, but since it isn't ready immediately, agreed that this retry is sub-optimal.
How about if we add this to the existing OSD health check [here](https://github.com/rook/rook/blob/master/pkg/operator/ceph/cluster/osd/health.go#L97)? Every time that health check queries the OSDs, it can also update the device class info if it changed.

Reply:
Using the existing OSD health check works for me. 👍

Reply:
Modified to add in the existing OSD health Check (avoided the retry mechanism) and moved the method `updateCephStorage` to `health.go`

Reply:
Modified to add in the existing OSD health check (avoided the retry mechanism) and moved the `updateCephStorage` to `health.go`